src/tally.cpp

```diff
@@ -440,14 +466,78 @@ std::string pyne::Tally::mcnp(int tally_index, std::string mcnp_version) {
     } else if (tally_type.find("Current") != std::string::npos) {
       // makes no sense in mcnp
     }
+  
+  
+  } else if (entity_type.find("Mesh") != std::string::npos) {
+    output << "FMESH4:" << particle_token << "  ";
```

Comment:
We should use a tally index - like in the other tallies - to allow multiple mesh tallies with different numbers.